data/data/openstack/masters/main.tf

```diff
@@ -40,21 +40,84 @@ resource "openstack_compute_servergroup_v2" "master_group" {
   policies = ["soft-anti-affinity"]
 }
 
-resource "openstack_compute_instance_v2" "master_conf" {
-  name = "${var.cluster_id}-master-${count.index}"
-  count = var.instance_count
+# The master servers are created in three separate resource definition blocks,
+# rather than a single block with a "count" meta-property, because they need to
+# be created sequentially rather than concurrently by Terraform.
+#
+# The reason why they need to be created one at a time is that OpenStack's
+# Compute module is currently unable to honour the "soft-anti-affinity" policy
+# when the servers are created concurrently.
+#
+# We chose to unroll the loop into three instances, because three is the
+# minimum number of required Control plane nodes, as stated in the
+# documentation[1].
+#
+# The expectation is that machine-api-operator will take care of creating any
```

Comment:
The control plane isn't managed by a MachineSet so MAO won't create any missing control nodes. I think we need a check in the validation logic to ensure that the control plane size is exactly `3`. Otherwise, won't the Cluster etcd Operator hang while waiting for nodes that will never be created?

Reply:
In my tests, the Installer creates as many Machine definitions as needed, and the additional nodes are being created in the cluster right after the first three.

Reply:
QE confirms: https://bugzilla.redhat.com/show_bug.cgi?id=1909587#c8

Reply:
Turns out @wjiangjay pointed me to the hard truth that we don't support installing more (nor less, actually) than exactly three masters: https://docs.openshift.com/container-platform/4.6/installing/installing_openstack/installing-openstack-installer-custom.html#installation-configuration-parameters_installing-openstack-installer-custom

It came as a surprise to me, and my test installation succeeded with 5 masters, with Terraform only creating the first three.

Reply:
(there's the same limitation on AWS, so I assume it's nothing platform-specific)